### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by saltsoftdrink** [MM2-0/Kvaesitso#616] [General]
Hey man just wanted to ask you which resources you used to learn kotlin & which videos will you refer to watch

**Comment by MM2-0** [comment]
Honestly, I don't remember. I started Android Dev in ~2015 (but before that I was already modding my phone using tutorials from XDA Developers). Back then, things used to be a lot different (and a lot simpler). Java was still the primary language for Android and we were still using the old View system. I starded like everyone started, with some simple apps (I don't remember what exactly, but I could check my hard drive if you want, I'm sure they are still lying around somewhere). And then I just never stopped. And whenever Google published new libraries, I tried to figure out how they worked and how to use them in my projects.

But to answer your specific question, I learned Kotlin reading some random Java to Kotlin migration guide. Don't remember which one, could be [this one](https://www.baeldung.com/kotlin/java-to-kotlin), idk. I knew Java fairly well at that point so learning the basics of Kotlin was only a matter of a few days. I would't recommend this as starting point though, if you don't know Java. And I also don't believe that you should learn Java first in order to learn Kotlin. It's just how I learned it.

Feel free to ask more questions if you have any.

**Comment by saltsoftdrink** [comment]
What is that XDA Developers?

**Reply by MM2-0** [reply]
It's an online forum about Android in general, but especially for modding. Nowadays it's mostly about custom roms (alternative Android versions that you can install on your device), but back then, when Android wasn't as polished as it is today, there were also a lot of tutorials for how to modify your existing system. Stuff like "how to change the color of [system component]" or "how to backport [feature that was introduced in Android 4.0] to Android 2.3". It usually involved decompiling system apps, adding some modifications, and then putting them back together. Fun days, but sadly not really a thing anymore (well to be fair, Android was pretty shit back then and I'm happy how it evolved)

**Comment by saltsoftdrink** [comment]
btw Thanks a Lot ❤️

**Comment by saltsoftdrink** [comment]
Just help me a Lil bit if you watch YT which channel's will you recommend to start/understand/get projects, according to you

**Reply by MM2-0** [reply]
I usually don't watch Youtube videos on that topic. I prefer written guides, because you can work in your own pace. Most of the times, I start with the official documentation. If I don't understand anything, I Google that question and there are usually several people on StackOverflow who had the same question and often there are also articles on Medium about the subject. I guess the most important skill is to break your problem into smaller problems until a problem is small enough that there are at least a dozen other people who had the same problem and have already found a solution for it.
